src/Pools/Pool.php

```diff
@@ -0,0 +1,197 @@
+<?php
+
+namespace Utopia\Pools;
+
+use Exception;
+
+class Pool
+{
+    /**
+     * @var string
+     */
+    protected string $name;
+    
+    /**
+     * @var int
+     */
+    protected $size = 0;
+
+    /**
+     * @var callable
+     */
+    protected $init = null;
+
+    /**
+     * @var int
+     */
+    protected $reconnectAttempts = 3;
+
+    /**
+     * @var int
+     */
+    protected $reconnectSleep = 1; // seconds
+
+    /**
+     * @var array
+     */
+    protected array $pool = [];
+
+    /**
+     * @var array
+     */
+    protected array $active = [];
+
+    /**
+     * @var string $name
+     * @var callable $init
+     */
+    public function __construct(string $name, int $size, callable $init)
+    {
+        $this->name = $name;
+        $this->size = $size;
+        $this->init = $init;
+    }
+
+    /**
+     * @return string
+     */
+    public function getName(): string
+    {
+        return $this->name;
+    }
+
+    /**
+     * @return int
+     */
+    public function getSize(): int
+    {
+        return $this->size;
+    }
+
+    /**
+     * @return int
+     */
+    public function getReconnectAttempts(): int
+    {
+        return $this->reconnectAttempts;
+    }
+
+    /**
+     * @return self
+     */
+    public function setReconnectAttempts(int $reconnectAttempts): self
+    {
+        $this->reconnectAttempts = $reconnectAttempts;
+        return $this;
+    }
+
+    /**
+     * @return int
+     */
+    public function getReconnectSleep(): int
+    {
+        return $this->reconnectSleep;
+    }
+
+    /**
+     * @return self
+     */
+    public function setReconnectSleep(int $reconnectSleep): self
+    {
+        $this->reconnectSleep = $reconnectSleep;
+        return $this;
+    }
+
+    /**
+     * @return self
+     */
+    public function fill(): self
+    {
+        $this->pool = [];
```

Comment:
Instad of an array, we can actually use a DataType that is built for this => https://www.php.net/manual/en/class.splqueue.php

Reply:
We can also use ArrayObject, but I think for this 2/3 operations we're using a simple array is good to keep it stupid.

Reply:
Just mentioned it, because instead of array push/pop/etc, there is dequeue and enqueue, which would translate better to our use case 👍🏻